### ISSUE TITLE
Reference node_name and check_validity in documentation.

### ISSUE THREAD

**Issue by rvanlaar** [IdentityPython/pysaml2#241]
I am using this metadata: https://engine.surfconext.nl/authentication/idp/metadata
The root node is EntityDescriptor, not Entit`ies`Descriptor.

This gives problems with the validation because the root node is not found. Changing this to the correct node_name fixes this issue.

It took me a long time to find and fix this. 
This problem also resulted in the error: "No IdP to send to given the premises"
Please include `node_name` and also the option `check_validity` in the documentation.

A solution is posted below.

```
'remote':  [
    {'url': 'https://engine.surfconext.nl/authentication/idp/metadata',
    'cert': os.path.join(SAML_ROOT, 'surf-cert.pem'),
    'node_name': 'urn:oasis:names:tc:SAML:2.0:metadata:EntityDescriptor'}]
```

I used pysaml 2.2.0.

**Comment by jkakavas** [comment]
I would rather we change the logic in https://github.com/rohe/pysaml2/blob/master/src/saml2/mdstore.py#L634 to handle the case where there is only one Entity in the metadata source . We seem to handle this in all other cases apart from signature validation

**Comment by peppelinux** [comment]
@c00kiemon5ter that's another reminder for the documentation. Very interesting

**Comment by bjm88** [comment]
this was very helpful and almost impossible to find in documentation.  Hoping similar node_name config works for hosting an IDP metadata, not just consuming an SP ?

**Comment by c00kiemon5ter** [comment]
Fixed by 4122e36 (try both `EntitiesDescriptor` and `EntityDescriptor`) and 204b582 (document `node_name` option)
